core/String.DecodeUTF8.savi

```diff
@@ -0,0 +1,440 @@
+    //  0 ----2---> 24 (one seen of `UTF8-2`)
+    // 24 --1,7,9-> 0  (`UTF8-2`, `UTF8-3`, or `UTF8-4` finished)
+    //  0 ---10---> 48 (one seen of `UTF8-3a`)
+    //  0 ----3---> 36 (one seen of `UTF8-3b`)
+    //  0 ----4---> 60 (one seen of `UTF8-3c`)
+    // 48 ----7---> 24 (two seen of `UTF8-3a`)
+    // 36 --1,7,9-> 24 (two seen of `UTF8-3b`)
+    // 60 ---1,9--> 24 (two seen of `UTF8-3c`)
+    //  0 ---11---> 72 (one seen of `UTF8-4a`)
+    //  0 ----6---> 84 (one seen of `UTF8-4b`)
+    //  0 ----5---> 96 (one seen of `UTF8-4c`)
+    // 72 ---7,9--> 36 (two seen of `UTF8-4a`)
+    // 84 --1,7,9-> 36 (two seen of `UTF8-4a`)
+    // 96 ----1---> 36 (two seen of `UTF8-4a`)
+    //
+    // We can also show the states and possible transitions as a graph:
+    //
+    ///////////////////////////////////////////////////////////////////
+    //                                                               //
+    //             2                                                 //
+    //          .------------------------.                           //
+    //         /                          \                          //
+    //        /    3                       \                         //
+    //       |   .--------------.           \                        //
+    //       |  /                \           \                       //
+    //       | /   5  .--. 1      \           |                      //
+    //   0   || .---->|96|------.  |          |                      //
+    //  .--. ||/      '--'       V V          V                      //
+    // |    \---.  6  .--. 1,7,9 .--. 1,7,9  .--. 1,7,9  .---.       //
+    //  '-->| 0 |---->|84|------>|36|------->|24|------->| 0 | . . . //
+    //      '---'     '--'       '--'        '--'        '---'       //
+    //       ||\  11  .--. 7,9    A          A A                     //
+    //       || '---->|72|--------'          | |                     //
+    //       | \      '--'                  /  |                     //
+    //       |  \ 10  .--. 7               /   |                     //
+    //       |   '--->|48|----------------'   /                      //
+    //        \       '--'                   /                       //
+    //         \   4  .--. 1,9              /                        //
+    //          '---->|60|-----------------'                         //
+    //                '--'                                           //
+    //                .--.                                           //
+    //      Note that |12| is the error state, which we will         //
+    //                '--'                                           //
+    //      reach if we encounter any edge not on this diagram.      //
+    //                                                               //
+    ///////////////////////////////////////////////////////////////////
+    //0  1   2   3   4   5   6   7   8   9  10  11 // byte classes
```

Comment:
@jemc I am just wondering if we could use a special syntax to make arrays of this kind more readable and less error-prone. Imagine array elements could be prefixed with their indices as in:

```savi
:const _state_table_data Array(U8)'val: [
   0:  0,  1: 12,  2: 24,  3: 36, ....., 11: 72
  12: 12,  ....
  24: 12,
]
```
Leaving of the prefixed index would +1 to the previous index. The compiler would warn or error if there are holes in the array literal or duplicates.

For the other lookup table, a Array(U4) would be nice :)